src/server/RequestHandler.cpp

```diff
@@ -284,7 +293,8 @@ bool RequestHandler::cgiProcessExited(const pid_t process_id, int &status) const
   pid_t re = waitpid(process_id, &status, WNOHANG);
   // errorまたはprocessが終了していない
   // errorのときの処理はあやしい, -1のエラーはロジック的にありえない(process idがおかしい)
-  if (re == 0 || re == -1) return false;
+  if (re == 0) return false;
+  // errorの時も子プロセスが存在しないと判断する
   return true;
```

Comment:
エラーを返すのはロジック的になさそうだと思ったのですが、どういうときに起こりうるかとかってわかりますか？

Reply:
kqueue serverでstress testを実行しているときに、現象として発生しました。
debugしてみると、process idは正しそうでした、
しかし、以下どちらかののno such a processみたいなエラーが発生していました。
```
       ECHILD (for wait()) The calling process does not have any
              unwaited-for children.

       ECHILD (for waitpid() or waitid()) The process specified by pid
              (waitpid()) or idtype and id (waitid()) does not exist or
              is not a child of the calling process.  (This can happen
              for one's own child if the action for SIGCHLD is set to
              SIG_IGN.  See also the Linux Notes section about threads.)
```

Reply:
もしこれを再現したかったら、元のコードに戻して、10000以上のクライアントで負荷テストを複数回してあげるとできるかもです。

Reply:
ありがとうございます。昨日m2macでやった時は起きなかったので聞かせてもらいました。
エラーが発生するのはどちらにしろwaitするプロセスがない時みたいなのでこれで大丈夫そうですね。